app/views/handlers/feedback.py

```diff
@@ -226,17 +234,48 @@ def __call__(self) -> Mapping:
 class FeedbackPayload:
     def __init__(
         self,
+        metadata,
+        session_store,
+        schema,
+        feedback_count,
         feedback_text,
         feedback_type,
         feedback_type_question_category=None,
     ):
+        self.metadata = metadata
+        self.session_store = session_store
+        self.schema = schema
+        self.feedback_count = feedback_count
         self.feedback_text = feedback_text
         self.feedback_type = feedback_type
         self.feedback_type_question_category = feedback_type_question_category
 
     def __call__(self) -> Mapping:
-        payload = vars(self)
+        payload = {
+            "origin": "uk.gov.ons.edc.eq",
+            "submitted_at": SubmissionHandler.submitted_at,
+            "collection": build_collection(self.metadata),
+            "metadata": build_metadata(self.metadata),
+            "survey_id": self.schema.json["survey_id"],
+            "tx_id": self.metadata["tx_id"],
+            "type": "uk.gov.ons.edc.eq:feedback",
+            "launch_language_code": self.metadata.get(
+                "language_code", DEFAULT_LANGUAGE_CODE
+            ),
+            "version": "0.0.3",
+        }
+
+        if self.metadata.get("form_type"):
+            payload["form_type"] = self.metadata["form_type"]
+
+        payload["data"] = {
```

Comment:
Rather than adding key and then deleting it later , why not conditionally update `payload` ?
```suggestion
        payload["data"] = {
            "feedback_text": self.feedback_text,
            "feedback_type": self.feedback_type,
            "feedback_count": self.feedback_count,
            }
          if self.feedback_type_question_category:
                    payload["data"].update({"feedback_type_question_category": self.feedback_type_question_category})
          return payload
```